platform: fix internal platform headers

0xc0170 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

Note: I am using platform as a component with CMake, therefore include paths are set differently than with our build system (CMake is not including everything thus this is failing).

If we take platform as a component, platform/source would be outside of this component. I moved internal headers into internal folder.

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [x] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

0xc0170 committed:
platform: remove platform/ prefix for source

If we take platform as a component, platform/source would be outside of this component.
I removed the prefix as these are implementation specific (they are located in source folder,
not in include as the others).

hugueskamba commented on platform/source/mbed_os_timer.cpp:
```diff
@@ -16,9 +16,9 @@
+#include "source/mbed_os_timer.h"
 #include "us_ticker_api.h"
 #include "lp_ticker_api.h"
 #include "mbed_critical.h"
```
How do you feel about the following in this same commit?
```suggestion
#include "platform/mbed_critical.h"
```

0xc0170 replied:
I rather do this separately, to get this in as it is.

0xc0170 committed:
platform: move internal headers to internal/

They belong to internal folder to follow our guideline, not in source as they were.

LDong-Arm commented on platform/source/mbed_os_timer.cpp:
```diff
@@ -16,9 +16,9 @@
-#include "platform/source/mbed_os_timer.h"
 #include "platform/CriticalSectionLock.h"
-#include "platform/source/SysTimer.h"
+#include "paltform/internal/SysTimer.h"
```
```suggestion
#include "platform/internal/SysTimer.h"
```

0xc0170 replied:
commited

0xc0170 committed:
Update platform/source/mbed_os_timer.cpp

fix the typo

Co-authored-by: Lingkai Dong <lingkai.dong@arm.com>

rajkan01 reviewed (approved):
LGTM
